### DISCUSSION TITLE
Resources are not loaded when using Octane

### DISCUSSION THREAD

**Discussion by VincentBean** [laravel/nova-issues#6458] [Q&A]
- Laravel Version: 11.14.0
- Nova Version: 4.34.0
- PHP Version: 8.3.7
- Database Driver & Version: n/a
- Operating System and Version: n/a
- Browser type and version: n/a
- Reproduction Repository: https://github.com/VincentBean/octane-nova-bug

### Description:

When using Octane, resources from packages are not loaded. They are not added to the menu and give a 404. When checking the contents of `Nova::$resources` it does not add them. Resources that exist in `app/Nova` do load via the `resourcesIn`.
Without Octane they do load and appear in the menu.

I've tested this on MacOS using Valet and on Ubuntu.

I have found one other mention on this issue by @rasmuscnielsen here: https://github.com/laravel/nova-issues/issues/6294#issuecomment-2020807845
But he never followed up with an issue.

### Detailed steps to reproduce the issue on a fresh Nova installation:

I've setup a repo with these two packages, which both register the Nova resource with `Nova::resource` in the service provider:
- https://github.com/bolechen/nova-activitylog
- https://github.com/justbetter/nova-error-logger

1. Clone & setup the repo
2. Create a user
3. Start Octane and login, the resources are not loaded
4. Login via FPM, the resources are loaded

**Comment by crynobone** [comment]
Octane only loads the service provider once in Octane while Nova added resources within Nova::serving() event so it only loaded when serving Nova requests.

**Comment by jmverges** [comment]
@VincentBean @crynobone I'm commenting here because today I found myself into the same scenario.

**Comment by VincentBean** [comment]
@crynobone So in other words, this will not be supported?

@jmverges Thank you! We have chosen to add an additional nginx rule so that Nova is handled via FPM:

```
    location ~ ^/(nova|nova-vendor|nova-api)(/.*)?$ {
        try_files $uri $uri/ /index.php?$query_string;
        fastcgi_split_path_info ^(.+\.php)(/.+)$;
        fastcgi_pass unix:/var/run/php/php8.3-fpm.sock;
        fastcgi_index index.php;
        include fastcgi_params;
    }
```

**Reply by crynobone** [reply]
> So in other words, this will not be supported?

What's I meant is you need to load the resources within `Nova::serving()` just like Laravel Nova application loads `resources()` method in `App\Providers\NovaServiceProvider`.

**Comment by jmverges** [comment]
@crynobone would be the same for Tool routes? I have the code like this and should be inside serving?
```
    public function boot()
    {
        $this->app->booted(function () {
            $this->routes();
        });

        Nova::serving(function (ServingNova $event) {
            //
        });

```

**Reply by crynobone** [reply]
No, routes must be loaded before any `Nova::serving()` gets resolved. I believe it be easier if you look inside `Laravel\Nova\NovaApplicationServiceProvider` and see what's get registered using `Nova::serving()` within the `boot()` method.

**Reply by jmverges** [reply]
I was checking it, so is my problem the fact those are inside $this->app->booted( in convination with Octane? Testing now here

**Reply by jmverges** [reply]
nop, it didn't work.
I have some routes like e/admin/resources/resource-name/6/custom-stuff
That works fine outside Octane and breaks with Octane enabled. I don't see anything wrong in the code and I could see is the same result
404
inside $this->app->booted or not.
I can see the routes when doing artisan route:list

**Reply by crynobone** [reply]
This seems to be unrelevant to the issue above. Why don't you create a new bug report with a reproducing repository.
